lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/InitMavenRequestTest.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat Inc. and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.lemminx.extensions.maven;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Properties;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(NoMavenCentralIndexExtension.class)
+public class InitMavenRequestTest {
+
+	@Test
+	public void testUserSettingsWithEnvVariables() throws Exception {
+		final File localSettingsFile = new File(getClass().getResource("/settings-local-repo-test.xml").toURI());
+		String localSettingsContent = FileUtils.readFileToString(localSettingsFile, "UTF-8");
+		String localRepoProperty = System.getProperty("maven.repo.local");
+		try {
+			Properties properties = System.getProperties();
+			properties.remove("maven.repo.local");
+			System.setProperties(properties);	// Not to use local repository set from properties
+			MavenLemminxExtension plugin = new MavenLemminxExtension();
+			plugin.settings.setGlobalSettings(null);
+			plugin.settings.setUserSettings(localSettingsContent);
+			
+			invoke(plugin, MavenLemminxExtension.class, "initialize", null, null);
+			
+			File localRepositoryPath = plugin.getMavenSession().getRequest().getLocalRepositoryPath();
+			assertNotNull(localRepositoryPath);
+			assertFalse(localRepositoryPath.getAbsolutePath().contains("${"));
```

Comment:
Test should verify the actual path is `System.getProperty("user.home") + ".m2/bug383repository"` instead of just checking for env-var expansion; so we know it's not a fallback strategy defalting to default repo (which would trigger a failure here).

Reply:
ACK